### PR TITLE
python3Packages.nbsphinx: 0.8.0 -> 0.8.1

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Fixes a [build failure on hydra](https://hydra.nixos.org/build/135943412). It looks like this environment variable needs to be set to run the tests during the build, or else something like `python.withPackages (ps: [ps.nbformat])'` would be need to be done during the build, rather than relying on propagatedBuildInputs to configure the environment correctly for testing, because it's looking for some /share paths that aren't set.

I did test that without setting any special environment variables, the package does work at runtime if you do something like

```
$ nix-shell -p '(import ./. {}).python38.withPackages (ps: [ps.nbsphinx])' --command "python -c 'import nbsphinx; print(nbsphinx)'"
```

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nbconflux/default.nix [hunk 1]
```diff
@@ -17,7 +25,7 @@ buildPythonPackage rec {
   checkInputs = [ pytest responses ];
 
   checkPhase = ''
-    pytest tests
+    JUPYTER_PATH=${nbconvert}/share/jupyter pytest tests -k 'not (test_post_to_confluence or test_optional_components)'
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nbconflux/default.nix:28]: Please use pytestCheckHook.

File: pkgs/development/python-modules/nbsphinx/default.nix [hunk 2]
```diff
@@ -30,7 +30,7 @@ buildPythonPackage rec {
   ];
 
   checkPhase = ''
-    ${python.interpreter} -m nbsphinx
+    JUPYTER_PATH=${nbconvert}/share/jupyter ${python.interpreter} -m nbsphinx
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nbsphinx/default.nix:33]: Can we use pythonImportsCheck with preCheck here?

File: pkgs/development/python-modules/nbconflux/default.nix (hunk 1 above)
Comment (rmcgibbo) [reply pkgs/development/python-modules/nbconflux/default.nix:28]: I can't get this env variable set during pytestCheckHook, and it fails without the env variable.

File: pkgs/development/python-modules/nbsphinx/default.nix (hunk 2 above)
Comment (rmcgibbo) [reply pkgs/development/python-modules/nbsphinx/default.nix:33]: Apparently not? It seems like an `export XYZ=123` in `checkPhase` doesn't change the environment for `pythonImportsCheckPhase`.

Comment (rmcgibbo) [reply pkgs/development/python-modules/nbsphinx/default.nix:33]: I can change it to
```
  checkPhase = ''
    export JUPYTER_PATH=${nbconvert}/share/jupyter
  '';
  pythonImportsCheck = [
    "nbsphinx"
  ];
```, which does work.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112056` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nbconflux</li>
<li>python37Packages.nbsphinx</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbsphinx</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbsphinx</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nbconflux</li>
<li>python37Packages.nbsphinx</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbsphinx</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbsphinx</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nbsphinx:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by rmcgibbo** [comment]
Thanks Sandro. I marked `nbsphinx` as `doCheck=false;` since it has no tests on github.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nbconflux</li>
<li>python37Packages.nbsphinx</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbsphinx</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbsphinx</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nbconflux</li>
<li>python37Packages.nbsphinx</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbsphinx</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbsphinx</li>
</ul>
</details>

**Pull request merged**